spatial/transform/umeyama.go

```diff
@@ -0,0 +1,140 @@
+// Copyright ©2025 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"errors"
+
+	"gonum.org/v1/gonum/mat"
+	"gonum.org/v1/gonum/stat"
+)
+
+// ErrSVDFailed is returned when the SVD factorization process fails.
+var ErrSVDFailed = errors.New("transform: SVD factorization failed")
+
+// Umeyama estimates the similarity transformation parameters between two matrices X and Y.
+//
+// This is an implementation of the algorithm presented in:
+// "Least-Squares Estimation of Transformation Parameters Between Two Point Patterns"
+// by Shinji Umeyama, IEEE Transactions on Pattern Analysis and Machine Intelligence,
+// Vol. 13, No. 4, April 1991, which can be found here: https://doi.org/10.1109/34.88573
```

Comment:
Suggest this goes at the end of the comment.
```suggestion
//
// "Least-Squares Estimation of Transformation Parameters Between Two Point Patterns"
// by Shinji Umeyama, IEEE Transactions on Pattern Analysis and Machine Intelligence,
// Vol. 13, No. 4, April 1991, [doi:10.1109/34.88573].
// [doi:10.1109/34.88573]: https://doi.org/10.1109/34.88573
```
Note for others, a free-access version exists [here](https://web.stanford.edu/class/cs273/refs/umeyama.pdf).